src/main/java/gyro/google/iam/ServiceAccountResource.java

```diff
@@ -0,0 +1,386 @@
+        this.email = email;
+    }
+
+    @Override
+    public void copyFrom(ServiceAccount model) throws Exception {
+        copyFrom(model, true);
+    }
+
+    @Override
+    protected boolean doRefresh() throws Exception {
+        Iam client = createClient(Iam.class);
+
+        ServiceAccount serviceAccount = client.projects()
+            .serviceAccounts()
+            .list(String.format("projects/%s", getProjectId()))
+            .execute()
+            .getAccounts().stream().filter(r -> r.getName().equals(getId())).findFirst().orElse(null);
+
+        if (serviceAccount == null) {
+            return false;
+        }
+
+        copyFrom(serviceAccount);
+
+        return true;
+    }
+
+    @Override
+    protected void doCreate(GyroUI ui, State state) throws Exception {
+        Iam client = createClient(Iam.class);
+
+        ServiceAccount serviceAccount = new ServiceAccount();
+        serviceAccount.setDisplayName(getDisplayName());
+
+        if (getDescription() != null) {
+            serviceAccount.setDescription(getDescription());
+        }
+
+        CreateServiceAccountRequest request = new CreateServiceAccountRequest();
+        request.setAccountId(getName());
+        request.setServiceAccount(serviceAccount);
+
+        ServiceAccount response = client.projects()
+            .serviceAccounts()
+            .create(String.format("projects/%s", getProjectId()), request)
+            .execute();
+
+        setId(response.getName());
+        state.save();
+
+        if (getEnableAccount() != null && getEnableAccount().equals(Boolean.FALSE)) {
+            changeServiceAccountStatus(client);
+        }
+
+        copyFrom(response, false);
+
+        if (!getPredefinedRoles().isEmpty() || !getCustomRoles().isEmpty()) {
+            manageIamPolicies();
+        }
+    }
+
+    @Override
+    protected void doUpdate(
+        GyroUI ui, State state, Resource current, Set<String> changedFieldNames) throws Exception {
+        Iam client = createClient(Iam.class);
+
+        if (changedFieldNames.contains("enable-account")) {
+            changeServiceAccountStatus(client);
+        }
+
+        if (changedFieldNames.contains("display-name")) {
+            ServiceAccount serviceAccount =
+                client.projects()
+                    .serviceAccounts()
+                    .get(getId())
+                    .execute();
+
+            serviceAccount.setDisplayName(getDisplayName());
+
+            client.projects().serviceAccounts().update(getId(), serviceAccount).execute();
+        }
+
+        if (changedFieldNames.contains("predefined-roles") || changedFieldNames.contains("custom-roles")) {
+            manageIamPolicies();
+        }
+    }
+
+    @Override
+    protected void doDelete(GyroUI ui, State state) throws Exception {
+        Iam client = createClient(Iam.class);
+
+        getCustomRoles().clear();
+        getPredefinedRoles().clear();
+
+        manageIamPolicies();
```

Comment:
Maybe a comment showing you are doing this to clear the iam roles
That is what this is doing right ?